### ISSUE TITLE
Template changes not reflected after re-generating the site

### ISSUE THREAD

**Issue by tomgreever** [bitovi/documentjs#178]
When I run `grunt documentjs` the template files seem to be built and stored in `node_modules/documentjs/site/static` and `node_modules/documentjs/site/templates` creating multiple folders with names like `75a21ea334caf26fc57508f75c447391`

However, when I make changes to my own template files in `/styles/theme/templates/layout.mustache` (for example) and generate the site again, the changes are often not reflected in the generated site. I must manually delete all the files in `node_modules/documentjs/site/templates` and generate the site in order to see my changes. 

I would expect `grunt documentjs` to automatically build the templates again instead of relying on the previous build.

**Comment by jandjorgensen** [comment]
@tomgreever This is what the '-f' flag is for http://documentjs.com/docs/DocumentJS.guides.generating.html

Does this just need to be added to the Live Style Guide documentation? Ideally, changing templates happens *mostly* during initial setup of a DocumentJS site. When actually working on documentation, most users probably don't want it to take the time to regenerate the static resources.

**Comment by tomgreever** [comment]
Thanks, I wasn't aware of the -f flag 

You're saying we cache those because on larger sites it would take too long to generate the pages? Can we instead detect changes in the static files to determine whether it needs to be built?

**Comment by justinbmeyer** [comment]
Not really. There's a process of overwriting defaults that happens. You would have to run that. And then you've just wasted some of the time you wanted to save. 

Although, -f is really about the static resource compilation. That takes much more time then templates.

**Comment by tomgreever** [comment]
Justin, which question of mine were you answering "not really" to?

**Comment by jandjorgensen** [comment]
It sounds like we can probably rebuild templates by default without a huge performance hit. I wonder if instead of using a "force" flag here we should just implement another *command* related to what you're trying to do.

e.g. MOST of the time, we're writing documentation and don't expect to have to think about the documentation site itself. SOME of the time, we're specifically changing the design of our documentation site. Instead of thinking of it as a "forced" operation, maybe we phrase it differently in the documentation and add a couple options (like `documentjs templates`).

I think from a mental load standpoint, it might make sense to assume that developers and designers should not be changing templates and static resources at the same time they're changing the actual documentation.
